src/desktop/components/auth_modal/view.coffee

```diff
@@ -113,6 +119,53 @@ module.exports = class AuthModalView extends ModalView
   toggleMode: (e) ->
     e.preventDefault()
     @state.set 'mode', $(e.target).data('mode')
+  
+  checkAcceptedTerms: () ->
+    input = $('input#accepted_terms_of_service').get(0)
+    input.setCustomValidity? ''
+    if $(input).prop('checked')
+      $('.tos-error').text ''
+      $boxContainer = $('.gdpr-signup__form__checkbox__accept-terms')
+      $boxContainer.attr('data-state', null)
+      true
+    else
+      $boxContainer = $('.gdpr-signup__form__checkbox__accept-terms')
+      $boxContainer.attr('data-state', 'error')
+      input = $('input#accepted_terms_of_service').get(0)
+      input.setCustomValidity('')
+      $('.tos-error').text 'Please agree to our terms to continue'
+      false
+
+  # showTosMessage: () ->
+  #   $boxContainer = $('.gdpr-signup__form__checkbox__accept-terms')
+  #   $boxContainer.attr('data-state', 'error')
+  #   input = $('input#accepted_terms_of_service').get(0)
+  #   input.setCustomValidity('')
+
+
+  fbSignup: (e) ->
+    e.preventDefault()
+    queryData =
+      'signup-intent': @signupIntent
+      'redirect-to': @currentRedirectTo()
+    queryString = $.param(queryData)
+    fbUrl = sd.AP.facebookPath + '?' + queryString
+    return window.location.href = fbUrl if @gdprDisabled
+
+    if @checkAcceptedTerms()
+      gdprString = $.param(@gdprData(@serializeForm()))
+      gdprFbUrl = fbUrl + "&" + gdprString
+      window.location.href = gdprFbUrl
+
+  # accomodate AB test for checkboxes
+  gdprData: (formData) ->
+    return {} if @gdprDisabled
+    if sd.GDPR_COMPLIANCE_TEST == 'separated_checkboxes'
```

Comment:
#minor coffeescript so u shd do `if sd.GDPR_COMPLICANTE_TEST is 'separated_checkboxes'` (and below too)